sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -55,26 +55,15 @@ def digest_hex_val(self) -> str:
         _, val = self.digest.split(":")
         return val
 
-    def purls(self, index_digest: Optional[str] = None) -> list[str]:
-        ans = []
-        if index_digest and self.arch:
-            ans.append(
-                PackageURL(
-                    type="oci",
-                    name=self.name,
-                    version=index_digest,
-                    qualifiers={"arch": self.arch, "repository_url": self.repository},
-                ).to_string()
-            )
-        ans.append(
-            PackageURL(
-                type="oci",
-                name=self.name,
-                version=self.digest,
-                qualifiers={"repository_url": self.repository},
-            ).to_string()
-        )
-        return ans
+    def purls(self, index_digest: str) -> list[str]:
+        purl = PackageURL(
+            type="oci",
+            name=self.name,
+            version=index_digest,
+            qualifiers={"arch": self.arch, "repository_url": self.repository},
```

Comment:
`self.arch` can be `None` here. PackageURL does the "right thing" and skips falsy values, but to avoid confusion / problems with misusing the API (`qualifiers` are annotated as a `dict[str, str]`), IMO we should only add arch when not None.